gamemode/modules/base/sh_util.lua

```diff
@@ -97,23 +97,25 @@ function DarkRP.findPlayers(info)
     local InfoPlayers = {}
     for A in string.gmatch(info .. ";", "([a-zA-Z0-9:_.]*)[;(,%s)%c]") do
         if A ~= "" then
-            table.insert(InfoPlayers, A)
+            InfoPlayers[#InfoPlayers+1] = A
```

Comment:
If you really want to optimise this, create an InfoPlayersCount variable and increment is inside the loop after each insertion so that the length doesn't have to be refetched on each iteration.

Reply:
Is `table.insert` really that slow that we should be avoiding it and doing this instead? At some point I prefer just using `table.insert` over doing manual accounting and accept the tiny cost it brings.

Reply:
See https://gitspartv.github.io/LuaJIT-Benchmarks/#test12 - this is a bit different of a case in application since the local count will be reused for a later loop.

Reply:
Look at the code though. This code splits the input string on semicolons and iterates over each element. I'm quite sure that this regex based split would already make the cost of `table.insert` completely invisible. On top of that, use of this semicolons syntax is extremely rare, so for most calls by far this is going to be a single iteration loop.

In DarkRP this function is only called for some chat commands. That's extremely rare performance wise. Let's not go to extreme lengths trying to optimize this function. It can only introduce bugs, make it less readable and for absolutely zero real world visible gain in performance.